meta-psa/README.md

```diff
@@ -65,16 +84,164 @@ and processed by the Linux kernel recipe used to build the kernel. For example,
+       In the example integration, the system user is `root`.
+* The FEKEK is generated, which will be wrapped and become the wrapped passphrase (i.e. the "wrapped FEKEK", see next points).
+* The top-level passphrase is generated and stored in `/home/root/.ecrypt`.
+* The FEKEK is encrypted (i.e. "wrapped FEKEK") with the top-level passphrase and stored in `/home/root/.ecrypt`.
+* Discard the FEKEK (i.e. the "unwrapped FEKEK").
+
+The following command can be used to wrap (encrypt) the FEKEK with the passphrase "passphrase",
+where `[wrapped FEKEK]` is the name of file containing the wrapped FEKEK:
+
+    # Wrapping the FEKEK
+    printf "%s\n%s" "FEKEK" "passphrase" | ecryptfs-wrap-passphrase [wrapped FEKEK] -
+
+#### 4.3.1.2 On each boot (including the first boot)
+The [psa-ecryptf-init.sh][psa-ecryptf-init.sh] script performs the following initialisation
+steps on the first and all subsequent boots:
+* Using the top-level passphrase from `/home/root/.ecrypt`, unwrap "wrapped FEKEK" and store it in the kernel user keyring.
+* Using the top-level passphrase from `/home/root/.ecrypt`, mount the encrypted storage in the unencrypted directory.
+
+The following command can be used to decrypt the wrapped FEKEK and insert it into the keyring,
+where `[wrapped FEKEK]` is the name of file containing the wrapped FEKEK:
+
+    # Insert FEKEK in keyring
+    printf "%s" "passphrase" | ecryptfs-insert-wrapped-passphrase-into-keyring [wrapped_FEKEK] -
+
+Once installed in the keyring, the FEKEK can be referenced with its signature (e.g.
+b89bed87989fcf3d). The following command mounts the encrypted files storage directory
+`/home/root/.secret` to make the decrypted files visible at the `/home/root/secret`
+mount point:
+
+    mount -t ecryptfs /home/root/.secret /home/root/secret -o \
+    ecryptfs_sig=b89bed87989fcf3d,ecryptfs_fnek_sig=b89bed87989fcf3d,\
+    ecryptfs_cipher=aes,ecryptfs_key_bytes=16,\
+    key=passphrase:passphrase_passwd_file=/home/root/.ecryptfs/passphrase.txt,\
+    ecryptfs_passthrough=n,ecryptfs_unlink_sigs,no_sig_cache
+
+### 4.3.2 Security concerns
+The top-level passphrase used to wrap the FEKEK is stored on the system under user account protection.
+This is discussed further in the next section.
+
+## <a name="section-4-4"></a> 4.4 Securing the eCryptfs top-level passphrase
+This section covers the eCryptfs top-level passphrase can be secured using platform specific hardware services.
+
+### 4.4.1 Generating the passphrase
+To be able to implement basic security of eCryptfs, there needs to be hardware support to
+generate a passphrase on each boot, so it's unnecessary to store the top-level passphrase in the file system.
+This passphrase needs to be different per device, but the same each boot as it will be used as the
+main eCryptfs passphrase that unwraps the "wrapped FEKEK" and mounts the encrypted storage.
+In the case that the platform does not contain a TPM (Trusted Platform Module) linked to the
```

Comment:
We should mention about what can be done with a TPM in the system, rather than suggesting it is so obvious no instructions are necessary.

Reply:
@jjohnson-arm Please take a look at the updated section with more TPM related details.